### ISSUE TITLE
Roles and RLS not visible in PBI Report Server

### ISSUE THREAD

**Issue by DasHammett** [otykier/TabularEditor#649]
**Describe the bug**
Roles and RLS created in Tabular Editor is not visible in a PowerBI Report Server. The Roles and RLS are visible and working as expected in a PBIX file.

**To Reproduce**
1. Create role and RLS in Tabular Editor
2. Save to PBIX and refresh 
3. Publish report to PowerBI Report Server
4. Add user to role -> role is not available

**Expected behavior**
Roles created in Tabular Editor should be available in PBI RS

**Observed behavior**
Created roles with a script as follows:
```
// Store name of role in a variable for easy reference:
var in_file = ReadFile(@"C:\Users\XXXX\Desktop\RLS.tsv");

// Split the file into rows by CR and LF characters:
var tsvRows = in_file.Split(new[] {'\r','\n'},StringSplitOptions.RemoveEmptyEntries);

foreach(var row in tsvRows) 
{
    var tsvColumns = row.Split(';'); 
    var roleName = tsvColumns[0];
    var comp_key = tsvColumns[1];

// Add the new role to the model:
Model.AddRole(roleName);

// Add a row level filter (DAX expression) on table 'Company' for the new role:
Model.Tables["Company"].RowLevelSecurity[roleName] = "[Company KEY] = \""+comp_key+"\"";

// Add permission to the role to Read, otherwise PBI_Desktop hangs trying to load
Model.Roles[roleName].ModelPermission = ModelPermission.Read;
}
```
No errors are generated and the model saves correctly to PBIX file.

**Application specifics**
 - Tabular Editor version: 2.12.4
 - Semantic Engine used: Power BI Desktop for RS May 2020 (2.81.5831.1181 (20.05))
 - Model Compatibility Level: Could not find info

**Additional context**
The only difference I see from roles created in PBI vs those created in Tabular Editor is that those created in PBI have a PBI_id as an annotation with a unique identifier.
![image](https://user-images.githubusercontent.com/13760961/95963386-ebe2c780-0e07-11eb-81d3-ecef806e6d3a.png)

As mentioned, the roles created with the above script load correctly in PBI Desktop and I can select the role and filters correctly the table (Company in the example), but once published in the Report Server, those are not visible and I cannot assign users

**Comment by otykier** [comment]
Roles are unfortunately [not currently supported for write operations by external tools](https://docs.microsoft.com/en-us/power-bi/create-reports/desktop-external-tools#supported-write-operations), so I'm afraid there's nothing I can do on the Tabular Editor side to fix this issue for now. There are many modelling features like this, which may seem to work fine in Power BI Desktop, but can cause issues when the PBIX is published elsewhere. That is why the list of supported write operations is restricted to measures, calc groups and perspectives for now.

Just out of curiosity - would it work if you manually added the `PBI_Id` annotation to the role you create as part of your script, or does it seem like Power BI creates a unique value for every role?

**Comment by DasHammett** [comment]
Hi, 

Adding manually the PBI_Id annotation to "1" makes the role available in the server. So I thought I would create fake PBI_Ids in the script. I then created a script that generated random numbers from 1 to 1000 and assigned one to each of the 50 roles I created.

When I tried to upload the PBIX to the server, it complained of a corrupted file. My *wild* guess is that if we were to know how this PBI_Id number is generated, we could perhaps replicate it in tabular editor.
